test/test_unary_ufuncs.py

```diff
@@ -1094,6 +1094,43 @@ def test_mish(self, device, dtype):
             rtol=rtol,
         )
 
+    @dtypes(torch.complex64, torch.complex128)
+    @onlyCPU
+    def test_log1p_complex(self, device, dtype):
+        inouts = [
+            (0.2 + 0.3j, 0.21263386770217202 + 0.24497866312686414j),
+            (1e-19 + 1e-18j, 1e-19 + 1e-18j),
+            (1e-18 + 0.1j, 0.00497517 + 0.0996687j),
+            (0.1 + 1e-18j, 0.0953102 + 9.090909090909090909e-19j),
+            (0.5 + 0j, 0.40546510810816 + 0j),
+            (0.0 + 0.5j, 0.111571776 + 0.463647609j),
```

Comment:
Consider using numpy as a ref check instead hard-coding the ref result?

Reply:
numpy's `log1p` suffers from loss of precision with relatively small complex number (see [here](https://github.com/numpy/numpy/pull/22611)). That's why I'm using the hard-coded results, calculated from arbitrary-precision math and wolfram alpha.

Reply:
Writing a test like this is so unusual now that, as @kit1980 mentions, it absolutely needs a comment. In the future we may want to extend the test framework to automatically generate "golden value" tests.